steps/terraform-precheck.yaml

```
@@ -0,0 +1,58 @@
+parameters:
+  - name: keyvaultName
+    default: ''
+
+  - name: keyvaultSecret
+    default: ''
+
+  - name: location
+    default: 'UK South'
+    values:
+      - 'UK South'
+      - 'UK West'
+
+  - name: serviceConnection
+    default: ''
+
+  - name: tfversion
```

don't shorten variables unless really needed,

```suggestion
  - name: terraformVersion
```